### PR TITLE
Integrate secure-boot check on ppc64le

### PR DESCRIPTION
**Pull request by Amrysliu**
Test secureboot  on Power:
Signed kernel and modules can be loaded, otherwise unsigned or 
signed with wrong key modules and kernel can't be loaded.

Related: https://jira.suse.com/browse/SLE-18851
Verify run:
  http://10.161.8.44/tests/9#step/bootloader/10
  http://10.161.8.44/tests/18

### CODE REVIEW

File: tests/security/secure_boot.pm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    assert_script_run("mkdir -p $work_dir");
+    assert_script_run("wget --quiet " . data_url($config_file) . " -O $cert_cfg");
+    assert_script_run("openssl req -new -nodes -utf8 -sha256 -days 36500 -batch -x509 -config $cert_cfg -outform DER -out $out_key -keyout $priv_key");
+
+    # Load module successfully when it has correct signature.
+    my $kernel_version = script_output('uname -r');
+    my $ufs_ko_file = "/lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko";
+    assert_script_run("zstd -d /lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko.zst");
+    validate_script_output("tail -c 28 /lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko | hexdump -C", sub { m/Module signatur/ });
```
Comment (lilyeyes) [inline tests/security/secure_boot.pm:40]: Is it "signatur" or "signature"?

Comment (lilyeyes) [inline tests/security/secure_boot.pm:40]: Why not use $ufs_ko_file to replace /lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko?

Comment (jouyingbin) [reply tests/security/secure_boot.pm:40]: and also should be **Module signature**?

File: tests/security/secure_boot.pm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    assert_script_run("insmod $ufs_ko_file");
+
+    assert_script_run('rmmod ufs');
+
+    # Fail to load unsigned module
+    assert_script_run("/usr/lib/rpm/pesign/kernel-sign-file sha256 $priv_key $out_key $ufs_ko_file");
```
Comment (lilyeyes) [inline tests/security/secure_boot.pm:46]: Could you please explain the purpose of this step ( "... kernel-sign-file sha256 $priv_key $out_key $ufs_ko_file" )?

**Comment by lilyeyes** [comment]
Added some comments and other code generally looks good to me.

**Comment by rfan1** [comment]
I suggest using the simple commit msg: 

Test secureboot on Power:
The signed kernel and modules could be loaded, otherwise unsigned or signed with wrong key
modules and kernel can't be loaded.

File: tests/security/secure_boot.pm [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Test the signed/unsigned module and kernel if can be loaded when secure-boot is enabled
+# Maintainer: Liu Xiaojing <xiaojing.liu@suse.com>
```
Comment (rfan1) [inline tests/security/secure_boot.pm:7]: Any test case number?

File: tests/security/secure_boot.pm [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    # generate key
+    my $work_dir = "/root/certs";
+    my $cert_cfg = "$work_dir/secure_boot_cert.conf";
+    my $priv_key = "$work_dir/key.pri";
+    my $out_key = "$work_dir/key.der";
+    my $config_file = 'openssl/gencert_conf/secure_boot_cert.conf';
+
+    assert_script_run("mkdir -p $work_dir");
+    assert_script_run("wget --quiet " . data_url($config_file) . " -O $cert_cfg");
+    assert_script_run("openssl req -new -nodes -utf8 -sha256 -days 36500 -batch -x509 -config $cert_cfg -outform DER -out $out_key -keyout $priv_key");
```
Comment (rfan1) [inline tests/security/secure_boot.pm:34]: @Amrysliu 

opensslca.pm may do the same thing, can you please double confirm? 
but not a big deal

File: tests/security/secure_boot.pm [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Load module successfully when it has correct signature.
+    my $kernel_version = script_output('uname -r');
+    my $ufs_ko_file = "/lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko";
+    assert_script_run("zstd -d /lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko.zst");
```
Comment (rfan1) [inline tests/security/secure_boot.pm:39]: assert_script_run q(zstd -d /lib/modules/`uname -r`/kernel/fs/ufs/ufs.ko.zst);

**Comment by jouyingbin** [comment]
@Amrysliu 
I think that **POWER** could be changed to **ppc64le**.

**Comment by hramrach** [comment]
related https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13528

File: tests/security/secure_boot.pm (hunk 1 above)
Comment (Amrysliu) [reply tests/security/secure_boot.pm:40]: It is `signatur`. 
Output looks like this `... |~Module signatur|`

File: tests/security/secure_boot.pm (hunk 3 above)
Comment (Amrysliu) [reply tests/security/secure_boot.pm:7]: no test case number, there is no poo for it.

File: tests/security/secure_boot.pm (hunk 4 above)
Comment (Amrysliu) [reply tests/security/secure_boot.pm:34]: There are some differences between them, so I prefer to keep it.

File: tests/security/secure_boot.pm (hunk 2 above)
Comment (Amrysliu) [reply tests/security/secure_boot.pm:46]: This is using `kernel-sign-file` to sign a module file using the given key.  It belongs to package `pesign-obs-integration`

**Review by rfan1 — APPROVED** [review]
lgtm

File: tests/security/secure_boot.pm (hunk 3 above)
Comment (lilyeyes) [reply tests/security/secure_boot.pm:7]: > no test case number, there is no poo for it.

Please open a poo then.

File: tests/security/secure_boot.pm [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    my ($self) = shift;
+    select_console 'root-console';
+
+    zypper_call('in pesign-obs-integration');
+
+    # Check if secure boot is enabled
+    validate_script_output('dmesg | grep -i secure', sub { m/Secure boot mode enabled/ });
+    validate_script_output('hexdump -C /proc/device-tree/ibm,secure-boot', sub { m/\d+\s+\d+\s+\d+\s+\d+\s+\d+/ });
+
+    # generate key
```
Comment (lilyeyes) [inline tests/security/secure_boot.pm:25]: s/g/G/

File: tests/security/secure_boot.pm [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    assert_script_run("mkdir -p $work_dir");
+    assert_script_run("wget --quiet " . data_url($config_file) . " -O $cert_cfg");
+    assert_script_run("openssl req -new -nodes -utf8 -sha256 -days 36500 -batch -x509 -config $cert_cfg -outform DER -out $out_key -keyout $priv_key");
+
+    # Load module successfully when it has correct signature.
+    my $kernel_version = script_output('uname -r');
+    my $ufs_ko_file = "/lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko";
+    assert_script_run("zstd -d /lib/modules/$kernel_version/kernel/fs/ufs/ufs.ko.zst");
+    validate_script_output("tail -c 28 $ufs_ko_file | hexdump -C", sub { m/Module signatur/ });
```
Comment (jouyingbin) [inline tests/security/secure_boot.pm:40]: Is this m/Module **signature**/ ?

Comment (Amrysliu) [reply tests/security/secure_boot.pm:40]: No, it's `sigatur`

File: tests/security/secure_boot.pm (hunk 6 above)
Comment (Amrysliu) [reply tests/security/secure_boot.pm]: fixed

**Pull request closed without merging**